google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataSettings.java

```diff
@@ -55,27 +56,57 @@
  * EnhancedBigtableStubSettings}, which is exposed as {@link Builder#stubSettings()}.
  */
 public final class BigtableDataSettings {
+
+  private static final Logger LOGGER = Logger.getLogger(BigtableDataSettings.class.getName());
+  private static final String BIGTABLE_EMULATOR_HOST_ENV_VAR = "BIGTABLE_EMULATOR_HOST";
+
   private final EnhancedBigtableStubSettings stubSettings;
 
   private BigtableDataSettings(Builder builder) {
     stubSettings = builder.stubSettings().build();
   }
 
-  /** Create a new builder. */
+  /**
+   * Create a new builder.
+   *
+   * <p>If emulator configuration provided in BIGTABLE_EMULATOR_HOST environment variable then it
+   * creates a builder preconfigured to connect to Bigtable using emulator hostname and port number.
+   */
   public static Builder newBuilder() {
+    String hostAndPort = System.getenv(BIGTABLE_EMULATOR_HOST_ENV_VAR);
+    if (hostAndPort != null && !hostAndPort.isEmpty()) {
+      try {
+        int lastIndexOfCol = hostAndPort.lastIndexOf(":");
+        int port = Integer.parseInt(hostAndPort.substring(lastIndexOfCol + 1));
+        return newBuilderForEmulator(hostAndPort.substring(0, lastIndexOfCol), port);
+      } catch (NumberFormatException ex) {
+        throw new RuntimeException(
+            "Invalid port in "
+                + BIGTABLE_EMULATOR_HOST_ENV_VAR
+                + " environment variable: "
+                + hostAndPort);
+      }
+    }
     return new Builder();
   }
 
-  /** Create a new builder preconfigured to connect to the Bigtable emulator. */
+  /** Create a new builder preconfigured to connect to the Bigtable emulator with port number. */
   public static Builder newBuilderForEmulator(int port) {
-    Builder builder = newBuilder();
+    return newBuilderForEmulator("localhost", port);
+  }
+
+  /**
+   * Create a new builder preconfigured to connect to the Bigtable emulator with o host & port name.
+   */
+  public static Builder newBuilderForEmulator(String hostname, int port) {
+    Builder builder = new Builder();
 
     builder
         .stubSettings()
         .setProjectId("fake-project")
         .setInstanceId("fake-instance")
```

Comment:
This diverges from the other clients. I don't think we should set project & instance ids any more